src/utils.cpp

```diff
@@ -178,6 +179,54 @@ StdioSilencer::~StdioSilencer()
   }
 }
 
+KConfig::KConfig()
+{
+  std::vector<std::string> config_locs;
+
+  // Try to get the config from BPFTRACE_KCONFIG_TEST env
+  // If not set, use the set of default locations
+  const char *path_env = std::getenv("BPFTRACE_KCONFIG_TEST");
+  if (path_env)
+    config_locs = { std::string(path_env) };
+  else
+  {
+    struct utsname utsname;
+    if (uname(&utsname) < 0)
+      return;
+    config_locs = {
+      "/boot/config-" + std::string(utsname.release),
+      "/proc/config.gz",
+    };
+  }
+
+  for (auto &path : config_locs)
+  {
+    // gzopen/gzgets handle both uncompressed and compressed files
+    gzFile file = gzopen(path.c_str(), "r");
+    if (!file)
+      continue;
+
+    char buf[4096];
+    while (gzgets(file, buf, sizeof(buf)))
+    {
+      std::string option(buf);
+      if (option.find("CONFIG_") == 0)
+      {
+        // trim trailing '\n'
+        if (option[option.length() - 1] == '\n')
+          option = option.substr(0, option.length() - 1);
+
+        auto split = option.find("=");
+        if (split == std::string::npos)
+          continue;
+
+        config.emplace(option.substr(0, split), option.substr(split + 1));
+      }
+    }
+    gzclose(file);
```

Comment:
Should there be a `break` after this? If both locations are available, probably best not to duplicate effort.